logaggregator/rfc5424/rfc5424.go

```diff
@@ -0,0 +1,172 @@
+package rfc5424
+
+import (
+	"bytes"
+	"errors"
+	"strconv"
+	"time"
+)
+
+type Message struct {
+	Facility       int
+	Severity       int
+	Version        int
+	Timestamp      time.Time
+	Hostname       string
+	AppName        string
+	ProcID         string
+	MsgID          string
+	StructuredData string
+	Msg            string
+}
+
+func (m Message) PriVal() int {
+	return m.Facility*8 + m.Severity
+}
+
+func Parse(buf []byte) (*Message, error) {
+	cursor := 0
+	msg := &Message{}
+	if err := parseHeader(buf, &cursor, msg); err != nil {
+		return nil, err
+	}
+
+	if err := parseStructuredData(buf, &cursor, msg); err != nil {
+		return nil, err
```

Comment:
Looks like these errors need a prefix. I'd suggest wrapping it in a error struct that includes the cursor so useful debug error messages can be constructed.